src/App.js

```diff
@@ -1,5 +1,135 @@
+  #menusMap;
+
+  #totalAmount = 0;
+  #discount = { christmas: 0, weekdays: 0, weekends: 0, special: false, isOffer: false };
+  #totalDiscountAmount = 0;
+  #finalAmount = 0;
+
+  async run() {
+    this.#date = await this.#getDate();
+    this.#menusMap = new Map(await this.#getMenus());
+    OutputView.printDate(this.#date);
+    OutputView.printMenu(this.#menusMap);
+    this.#totalAmount = this.#calculateTotalAmount();
+    OutputView.printTotalAmount(this.#totalAmount);
+    this.#discount.isOffer = this.#isOffer();
+    OutputView.printOffer(this.#discount.isOffer);
+    this.#getDiscount();
+    OutputView.printDiscount(this.#discount);
+    this.#totalDiscountAmount = this.#calculateTotalDiscountAmount();
+    OutputView.printTotalDiscountAmount(this.#totalDiscountAmount);
+    this.#finalAmount = this.#calculateFinalAmount();
+    OutputView.printFinalAmount(this.#finalAmount);
+    OutputView.printEventBadge(this.#totalDiscountAmount);
+  }
+
+  /** 날짜를 입력받고 this.#date에 날짜를 number형으로 저장합니다. */
+  async #getDate() {
+    try {
+      const date = await InputView.readDate();
+      Validation.date(date);
+      return Number(date);
+    } catch (error) {
+      Console.print(error.message);
+      await this.#getDate();
+    }
+  }
+
+  /** 주문할 메뉴를 입력받고 this.#menusMap에 메뉴를 Map<메뉴, 개수> 형식으로 저장합니다. */
+  async #getMenus() {
+    try {
+      const menus = await InputView.readMenus();
+      Validation.menus(menus);
+      return menus.map((menu) => menu.split("-"));
+    } catch (error) {
+      Console.print(error.message);
+      await this.#getMenus();
+    }
+  }
+
+  /** 주문 받은 메뉴를 기반으로 할인 전 총 주문금액을 구합니다. */
+  #calculateTotalAmount() {
+    // 메뉴판의 모든 메뉴와 가격을 객체 형태로 새롭게 생성
+    const allMenuPrice = {};
+    Object.values(MENU).forEach((category) =>
+      Object.entries(category).forEach(([_i, { name, price }]) => (allMenuPrice[name] = price))
+    );
+
+    let totalAmount = 0;
+    this.#menusMap.forEach((count, dish) => {
+      totalAmount += allMenuPrice[dish] * count;
+    });
+    return totalAmount;
+  }
+
+  /**
+   * 증정 이벤트 여부를 구합니다.
+   * @returns {boolean}
+   */
+  #isOffer() {
+    if (this.#totalAmount >= OFFER_MENU.applyPoint) return true;
+    return false;
+  }
+
+  /** 적용될 할인 혜택을 구하고 this.#discount 객체에 담습니다. */
+  #getDiscount() {
+    if (this.#discount <= EVENT_DAYS.CHRISTMAS) {
+      this.#discount.christmas = EVENT.CHRISTMAS.calculate(this.#date);
+    }
+
+    if (EVENT_DAYS.WEEKENDS.includes(this.#date)) {
+      const mains = MENU.main.map((item) => item.name);
+      const count = this.#calculateWeekDiscount(mains);
+      this.#discount.weekends = EVENT.WEEKENDS.calculate(count);
+    } else {
+      const desserts = MENU.dessert.map((item) => item.name);
+      const count = this.#calculateWeekDiscount(desserts);
+      this.#discount.weekdays = EVENT.WEEKDAYS.calculate(count);
+      if (EVENT_DAYS.SPECIAL.includes(this.#date)) this.#discount.special = true;
+    }
```

Comment:
```suggestion
    if (EVENT_DAYS.WEEKENDS.includes(this.#date)) {
      const mains = MENU.main.map((item) => item.name);
      const count = this.#calculateWeekDiscount(mains);
      this.#discount.weekends = EVENT.WEEKENDS.calculate(count);
      return;
    }
    
      const desserts = MENU.dessert.map((item) => item.name);
      const count = this.#calculateWeekDiscount(desserts);
      this.#discount.weekdays = EVENT.WEEKDAYS.calculate(count);
      
      if (EVENT_DAYS.SPECIAL.includes(this.#date)) this.#discount.special = true;
```
조건이 너무 많고 해서 가독성이 떨어지는 것 같네요... `early return`을 사용해봐도 좋을 것 같아요!

Reply:
오호! 이 경우에는 `early return`이 가독성에 도움이 되는 군요! 헷갈리던 부분이었는데 감사합니다 ☺️